packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -203,12 +203,15 @@ export function getScopeTokenClass(owner: VM): string | null {
 /**
  * This function returns the host style token for a custom element if it
  * exists. Otherwise it returns null.
+ *
+ * A host style token is applied to the component if scoped styles are used.
  */
 export function getStylesheetTokenHost(vnode: VCustomElement): string | null {
-    const {
-        template: { stylesheetToken },
-    } = getComponentInternalDef(vnode.ctor);
-    return !isUndefined(stylesheetToken) ? makeHostToken(stylesheetToken) : null;
+    const { template } = getComponentInternalDef(vnode.ctor);
+    const { stylesheetToken } = template;
+    return computeHasScopedStyles(template) && !isUndefined(stylesheetToken)
```

Comment:
```suggestion
    return !isUndefined(stylesheetToken) && computeHasScopedStyles(template)
```

Avoids an expensive computation if the `stylesheetToken` is undefined.